src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -312,6 +331,16 @@ protected override Encoding StandardOutputEncoding
 
         protected override int ExecuteTool(string pathToTool, string responseFileCommands, string commandLineCommands)
         {
+            if (ProvideCommandLineInvocation)
+            {
+                CommandLineInvocation = GenerateResponseFileContents();
```

Comment:
So it seems MSBuild ToolTasks commonly throw around entire strings for this. Case in point, just below this addition, the entire commandLineCommands string is only divided into other strings when it's being sent to the resident server compiler.

However, I like the idea of a `string[]` of arguments better anyway. I've prepared such a change.